yarn-project/validator-client/src/validator.ts

```diff
@@ -226,6 +226,13 @@ export class ValidatorClient extends (EventEmitter as new () => WatcherEmitter)
     const blockNumber = proposal.blockNumber;
     const proposer = proposal.getSender();
 
+    // Check that I have any address in current committee before attesting
+    const inCommittee = await this.epochCache.filterInCommittee(this.keyStore.getAddresses());
+    if (inCommittee.length === 0) {
+      this.log.verbose(`No validator in the committee, skipping attestation`);
+      return undefined;
+    }
+
```

Comment:
Move this check at the top of the function in order for the 'failed attestations' metrics to not be disturbed if the validator is not on the current committee